### PR TITLE
fixes ImageNet, SimpleRNN examples

### PR DESCRIPTION
**Pull request by DhruvDh**
Fixes https://github.com/LuxDL/Lux.jl/issues/498

Running the `resnet` model per the example from the ImageNet README requires adding `Flux` and `Metalhead` per the `Blotz` README. I added those and replicated the `unsafe_free!` for AMDGPU.

I am new to Julia so sorry if something isn't quite right.

Also added the fix to #498 mentioned in the conversation.

### CODE REVIEW

File: examples/ImageNet/main.jl [hunk 1]
```diff
@@ -62,7 +64,8 @@ function construct(rng::AbstractRNG, cfg::ModelConfig, ecfg::ExperimentConfig)
 end
 
 function construct(cfg::OptimizerConfig)
-    if cfg.name == "adam" + opt = Adam(cfg.learning_rate)
+    if cfg.name == "adam"
```
Comment (DhruvDh) [inline examples/ImageNet/main.jl:67]: this did not compile for me, and I wrote what I assumed was intended.

File: examples/GravitationalWaveForm/main.jl [hunk 2]
```diff
@@ -17,7 +17,9 @@ close(pkg_io) #hide
 using Lux, ComponentArrays, LineSearches, LuxAMDGPU, LuxCUDA, OrdinaryDiffEq, Optimization,
       OptimizationOptimJL, Random, SciMLSensitivity
 using CairoMakie, MakiePublication
+
 CUDA.allowscalar(false)
+AMDGPU.allowscalar(false)
```
Comment (avik-pal) [inline examples/GravitationalWaveForm/main.jl:22]: This shouldn't be needed. CUDA sets the one in GPUArrays I think.

File: examples/Basics/main.jl [hunk 3]
```diff
@@ -117,9 +117,13 @@ W * x
 # above operations with the same syntax.
 
 using LuxCUDA
+using LuxAMDGPU
 if LuxCUDA.functional()
     x_cu = cu(rand(5, 3))
     @show x_cu
+elseif LuxAMDGPU.functional() # Similarly, for AMDGPU
+    x_amd = gpu(rand(5, 3))
```
Comment (avik-pal) [inline examples/Basics/main.jl:125]: `roc` not `gpu`

**Review by avik-pal — CHANGES_REQUESTED** [review]
Remove the `AMDGPU.allowscalar` and we should be good.

File: examples/ImageNet/main.jl [hunk 4]
```diff
@@ -2,6 +2,8 @@
 
 using Augmentor                                         # Image Augmentation
 using Boltz                                             # Computer Vision Models
+using Flux                                              # Some Blotz Models need Flux
+using Metalhead                                         # Some Blotz Models need MetalHead
```
Comment (avik-pal) [inline examples/ImageNet/main.jl:6]: make these `import` else there will be too many export clashed

File: examples/SimpleRNN/main.jl [hunk 5]
```diff
@@ -119,7 +119,7 @@ function compute_loss(x, y, model, ps, st)
     return binarycrossentropy(y_pred, y), y_pred, st
 end
 
-matches(y_pred, y_true) = sum((y_pred .> 0.5) .== y_true)
+matches(y_pred, y_true) = sum((y_pred .> 0.5f0) .== y_true)
```
Comment (DhruvDh) [inline examples/SimpleRNN/main.jl:122]: changed this for Metal compatibility, apple GPUs don't seem to support float64

File: examples/SimpleRNN/main.jl [hunk 6]
```diff
@@ -16,7 +16,7 @@ Pkg.instantiate(; io=pkg_io) #hide
 Pkg.develop(; path=joinpath(__DIR, "..", ".."), io=pkg_io) #hide
 Pkg.precompile(; io=pkg_io) #hide
 close(pkg_io) #hide
-using Lux, LuxAMDGPU, LuxCUDA, JLD2, MLUtils, Optimisers, Zygote, Random, Statistics
+using Lux, LuxAMDGPU, LuxCUDA, Metal, JLD2, MLUtils, Optimisers, Zygote, Random, Statistics
```
Comment (avik-pal) [inline examples/SimpleRNN/main.jl:19]: Dont load `Metal` it is not widely supported in Lux anyways

File: examples/SimpleRNN/main.jl (hunk 5 above)
Comment (avik-pal) [reply examples/SimpleRNN/main.jl:122]: sounds good

**Comment by avik-pal** [comment]
Our GPU CIs are down atm and I don't think are going to be online before the next week, so we will have to wait till then to merge

File: examples/ImageNet/main.jl [hunk 7]
```diff
@@ -2,6 +2,8 @@
 
 using Augmentor                                         # Image Augmentation
 using Boltz                                             # Computer Vision Models
+import Flux                                              # Some Blotz Models need Flux
+import Metalhead                                         # Some Blotz Models need MetalHead
```
Comment (avik-pal) [inline examples/ImageNet/main.jl:6]:
```suggestion
import Flux                                             # Some Blotz Models need Flux
import Metalhead                                        # Some Blotz Models need MetalHead
```

File: examples/Basics/main.jl [hunk 8]
```diff
@@ -117,9 +117,13 @@ W * x
 # above operations with the same syntax.
 
 using LuxCUDA
+using LuxAMDGPU
```
Comment (avik-pal) [inline examples/Basics/main.jl:120]:
```suggestion
using LuxCUDA, LuxAMDGPU

```

**Comment by avik-pal** [comment]
Thanks for the fixes, merged it via the other PR

**Pull request closed without merging**